Modix/ClientApp/src/components/UserLookup/UserSearch.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+    <section class="section">
+        <div class="message-narrow">
+            <div class="field">
+                <label class="label">User ID</label>
+                <div class="control">
+                    <input :class="inputClass" type="text" placeholder="Enter a User ID..." v-model="userId" />
+                </div>
+            </div>
+        </div>
+    </section>
+</template>
+
+<script lang="ts">
+import { Component, Watch } from 'vue-property-decorator';
+import ModixComponent from '@/components/ModixComponent.vue';
+import UserService from '@/services/UserService';
+
+@Component({})
+export default class UserSearch extends ModixComponent
+{
+    userId: string | null = null;
+
+    inputClass: any = 'input';
+
+    @Watch('userId')
+    async computeInputClass(): Promise<void>
+    {
+        if (this.userId == null || this.userId.length == 0)
+        {
+            this.inputClass = 'input';
+            return;
+        }
+
+        let user = await UserService.getUserInformation(this.userId);
+
+        if (!user || user.id <= 0)
```

Comment:
I don't see how `user.id` could be anything other than `null` or `EphemeralUser` here (because that's what the promise returns), the user either exists or it doesn't, i.e. it's unnecessary to check `user.id <= 0`, no?

With that said, this is insanely picky, so feel free to ignore 👍

Reply:
You just made me realize that `user.id` shouldn't even be a `number`, since Discord IDs don't fit in `number`. So that check should become `if (!user || user.id.length == 0)` I think

Reply:
And I just realized I typed `user.id` when I meant `user`.
Glad I helped you out though, I wouldn't have caught that since I don't know much about vue 👍

Reply:
Regardless, I still believe the check for `user.id` is completely unnecessary since we check if it's a valid ulong on the backend and return null if it isn't

Reply:
Fair enough